cpp/tests/test_abm_world.cpp

```diff
@@ -299,46 +299,50 @@ TEST(TestWorld, evolveMigration)
-        auto home_id = world.add_location(mio::abm::LocationType::Home);
-        auto work_id = world.add_location(mio::abm::LocationType::Work);
-        auto icu_id  = world.add_location(mio::abm::LocationType::ICU);
+        auto world       = mio::abm::World(params);
+        auto home_id     = world.add_location(mio::abm::LocationType::Home);
+        auto work_id     = world.add_location(mio::abm::LocationType::Work);
+        auto icu_id      = world.add_location(mio::abm::LocationType::ICU);
+        auto hospital_id = world.add_location(mio::abm::LocationType::Hospital);
         // Create a person that is dead at time t - dt
-        auto& p_dead =
-            add_test_person(world, icu_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t - dt);
+        auto& p_dead = add_test_person(world, icu_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t);
         // Create a person that will be dead at time t
         auto& p_severe =
-            add_test_person(world, home_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t);
+            add_test_person(world, hospital_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t + dt);
         p_dead.set_assigned_location(icu_id);
+        p_dead.set_assigned_location(work_id);
+        p_dead.set_assigned_location(home_id);
+        p_severe.set_assigned_location(hospital_id);
+        p_severe.set_assigned_location(icu_id);
         p_severe.set_assigned_location(home_id);
 
         // Add trip to see if a dead person can move outside of cemetery by scheduled
-        mio::abm::TripList& data = world.get_trip_list();
+        mio::abm::TripList& trip_list = world.get_trip_list();
         mio::abm::Trip trip1(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(2), work_id, home_id);
         mio::abm::Trip trip2(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(3), home_id, icu_id);
         mio::abm::Trip trip3(p_severe.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(3), home_id, icu_id);
-        data.add_trip(trip1);
-        data.add_trip(trip2);
-        data.add_trip(trip3);
+        trip_list.add_trip(trip1);
+        trip_list.add_trip(trip2);
+        trip_list.add_trip(trip3);
 
-        // Check the dead person got burried and the servere person starts in Hospital
-        world.evolve(t - dt, dt);
+        // Check the dead person got burried and the severely infected person starts in Hospital
+        world.evolve(t, dt);
         EXPECT_EQ(p_dead.get_location().get_type(), mio::abm::LocationType::Cemetery);
-        EXPECT_EQ(p_severe.get_infection_state(t - dt), mio::abm::InfectionState::InfectedSymptoms);
-        EXPECT_EQ(p_severe.get_location().get_type(), mio::abm::LocationType::Home);
+        EXPECT_EQ(p_severe.get_infection_state(t), mio::abm::InfectionState::InfectedSymptoms);
+        EXPECT_EQ(p_severe.get_location().get_type(), mio::abm::LocationType::Hospital);
```

Comment:
Why is the person with `InfectedSymptoms` (which is not `InfectedSevere`) already in hospital?

Reply:
Yes, the location is not directly handled by the `InfectionState`, the `MigrationRules` ensure this.

However, I do not perfectly understand why your person was in `InfectedSymptoms` before. You rolled the infection states backwards, if it is in `Dead` state in t+dt, you went back 0.01 from Dead to Critical and 0.01 from Critical to Severe but SymptomsToSevere has probably been larger than dt-2*0.01, right?

A better approach anyway would be to chose `SevereToCritical+CriticalToDead < 2dt`.

Reply:
I think my previous comment was not correct. I think I understand why the person was in `InfectedSymptoms`. Please see my other comment below and reintroduce the setting of `SevereToCritical` to avoid problems due to default parameter setting.